tendermint/src/timeout.rs

```diff
@@ -32,28 +32,35 @@ impl FromStr for Timeout {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
+        enum Unit {
+            Milliseconds,
+            Seconds,
+        }
+
         // Timeouts are either 'ms' or 's', and should always end with 's'
         if s.len() < 2 || !s.ends_with('s') {
             return Err(err!(ErrorKind::Parse, "invalid units"));
         }
 
         #[allow(clippy::wildcard_enum_match_arm)]
         let units = match s.chars().nth(s.len() - 2) {
-            Some('m') => "ms",
-            Some('0'..='9') => "s",
+            Some('m') => Unit::Milliseconds,
+            Some('0'..='9') => Unit::Seconds,
             _ => return Err(err!(ErrorKind::Parse, "invalid units")),
         };
 
-        let numeric_part = s.chars().take(s.len() - units.len()).collect::<String>();
+        let numeric_part = match units {
```

Comment:
What does this enum have to do with `Rework use of unreachable` ?

Also, can the Unit variants contain their own string representation so we don't need a match here like before?